packages/grid/test/resizing.test.js

```diff
@@ -213,4 +215,73 @@ describe('all rows visible', () => {
       expect(getPhysicalItems(grid).length).to.be.above(0);
     });
   });
+
+  describe('max-height', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      wrapper = fixtureSync(`
+        <div>
+          <vaadin-grid>
+            <vaadin-grid-column path="value"></vaadin-grid-column>
+          </vaadin-grid>
+        </div>
+      `);
+      grid = wrapper.firstElementChild;
+      grid.allRowsVisible = true;
+      grid.style.maxHeight = '300px';
+      grid.items = [...Array(100)].map((_, idx) => ({ value: idx }));
+      grid.dataProvider = sinon.spy(grid.dataProvider);
+      flushGrid(grid);
+    });
+
+    it('should include rows', () => {
```

Comment:
Could it be a visual test?

Reply:
I considered it but since this isn't a theme-related change, a visual test would be unnecessarily heavy for the purpose.

Reply:
Oh, one additional benefit of testing here is that it ensures the change works on all the supported browsers.